ci-operator/step-registry/aks/provision/aks-provision-commands.sh

```diff
@@ -3,13 +3,29 @@
 set -euo pipefail
 
 AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+if [[ "${USE_HYPERSHIFT_AZURE_CREDS}" == "true" ]]; then
+    AZURE_AUTH_LOCATION="/etc/hypershift-ci-jobs-azurecreds/credentials.json"
+fi
 AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
 AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
 AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
-AZURE_LOCATION="$LEASED_RESOURCE"
+AZURE_LOCATION="${HYPERSHIFT_AZURE_LOCATION:-${LEASED_RESOURCE}}"
 
 RESOURCE_NAME_PREFIX="${NAMESPACE}-${UNIQUE_HASH}"
 
+CLUSTER_AUTOSCALER_ARGS=""
+if [[ "${ENABLE_CLUSTER_AUTOSCALER:-}" == "true" ]]; then
+    CLUSTER_AUTOSCALER_ARGS="--enable-cluster-autoscaler"
+fi
+
+if [[ "${AKS_CLUSTER_AUTOSCALER_MIN_NODES:-}" != "" ]]; then
+    CLUSTER_AUTOSCALER_ARGS+=" --min-count ${AKS_CLUSTER_AUTOSCALER_MIN_NODES}"
+fi
+
+if [[ "${AKS_CLUSTER_AUTOSCALER_MAX_NODES:-}" != "" ]]; then
+    CLUSTER_AUTOSCALER_ARGS+=" --max-count ${AKS_CLUSTER_AUTOSCALER_MAX_NODES}"
+fi
```

Comment:
I'm wondering what happens if `--min-count` and/or `--max-count` are specified when `ENABLE_CLUSTER_AUTOSCALER` is false. 

No objections to the current implementation though.

Reply:
They're all required so the command would fail if any of them were missing.